### PR TITLE
#54 combined etl script

### LINKED ISSUE #54: Create combined ETL script
Description:
Create a combined script.

Outcome:
A fully-functioning etl pipeline script.

User Story:
As a data engineer, I want to create a combined ETL pipeline script so that I can automate the process of extracting, transforming, and loading data into the desired destination.

### PR DESCRIPTION
**Pull request by keoghrmj**
A combined file to run the whole pipeline has been created (`etl.py`). In addition to this, the dockerfile has been updated to run the whole pipeline via this new file.

Closes #54 and #58

### CODE REVIEW

File: pipeline/etl.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+        with get_connection() as conn:
+            botanist_names = metric_df['name'].unique().tolist()
+
+            botanist_id_mapping = get_botanists_id_mapping(
+                conn, botanist_names)
+
+            insert_into_plant_metric(conn, metric_df, botanist_id_mapping)
+
+    except Exception as e:
```
Comment (SurinaCS) [inline pipeline/etl.py:37]: is there a more specific error you could add?

File: pipeline/etl.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        metric_df = plant_metrics
+
+        with get_connection() as conn:
+            botanist_names = metric_df['name'].unique().tolist()
+
+            botanist_id_mapping = get_botanists_id_mapping(
+                conn, botanist_names)
+
+            insert_into_plant_metric(conn, metric_df, botanist_id_mapping)
+
```
Comment (SurinaCS) [inline pipeline/etl.py:36]: very succinct

**Review by SurinaCS — CHANGES_REQUESTED** [review]
nice work, maybe add a specific error

File: pipeline/etl.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+from extract import fetch_and_collect_data
+from transform import convert_datatypes, round_floats, remove_punctuation, verify_emails
+from load import get_botanists_id_mapping, get_connection, insert_into_plant_metric
+
+DECIMAL_PLACES = 2
+EMAIL_REGEX = """(?:[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*|\"(?:[\\x01-\\x08\\x0b\\x0c\\x0e-\\x1f\\x21\\x23-\\x5b\\x5d-\\x7f]|\\\\[\\x01-\\x09\\x0b\\x0c\\x0e-\\x7f])*\")@(?:(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?|\\[(?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\\.){3}(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?|[a-z0-9-]*[a-z0-9]:(?:[\\x01-\\x08\\x0b\\x0c\\x0e-\\x1f\\x21-\\x5a\\x53-\\x7f]|\\\\[\\x01-\\x09\\x0b\\x0c\\x0e-\\x7f])+)\\])"""
+
+
+def main() -> None:
+    """Runs the main functions for the ETL Pipeline"""
```
Comment (FarihaChoudhury) [inline pipeline/etl.py:16]: what does the ETL pipeline do?

File: pipeline/etl.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        with get_connection() as conn:
+            botanist_names = metric_df['name'].unique().tolist()
+
+            botanist_id_mapping = get_botanists_id_mapping(
+                conn, botanist_names)
+
+            insert_into_plant_metric(conn, metric_df, botanist_id_mapping)
+    except exceptions.OperationalError as e:
+        logging.error("Failed to connect to the database: %s", e)
+        raise
```
Comment (SurinaCS) [inline pipeline/etl.py:45]: this is much better

File: pipeline/extract.py [hunk 5]
```diff
@@ -89,7 +89,7 @@ async def fetch_plant_data(session: aiohttp.ClientSession, number: int) -> dict:
         return None
 
 
-async def fetch_and_collect_data() -> list:
+async def fetch_and_collect_data() -> list[dict]:
```
Comment (SurinaCS) [inline pipeline/extract.py:92]: better type hints here

**Review by SurinaCS — APPROVED** [review]

**Pull request merged**